Users/middleware/verify_jwt.js

```diff
@@ -0,0 +1,19 @@
+import jwt from 'jsonwebtoken';
+const TOKEN_KEY = '' + process.env.SECRET_KEY;
+
+const authenticateToken = async (req, res, next) => {
+  if (req.headers.authorization) {
+    const token = req.headers.authorization.split('Bearer ')[1];
+    jwt.verify(token, TOKEN_KEY, (err) => {
+      if (err) {
+        res.status(401).json({ error: 'AUTH_ERROR' });
+      } else {
+        next();
+      }
+    });
+  } else {
+    res.status(401).json({ error: 'AUTH_ERROR_FROM_AUTHCHECKER' });
+  }
+};
```

Comment:
```suggestion
  if (req.headers.authorization) {
    const token = req.headers.authorization.split('Bearer ')[1];
    jwt.verify(token, TOKEN_KEY, (err) => {
      if (err) {
        res.status(401).json({ error: 'UNAUTHORIZED' });
      } else {
        next();
      }
    });
  } else {
    res.status(401).json({ error: 'USER_TOKEN_REQUIRED' });
  }
};
```
에러 메세지는 프론트 개발자가 문제를 해결하기에 적절해야합니다 :) 따라서 그냥 auth check에서 에러났다기 보다는 명시해서 알려주시는 것이 좋습니다.

에러메세지를 두루뭉술하게 작성하는 경우는 회원가입과 로그인에 한해서만 적용하시면 됩니당

Reply:
네! 감사합니다. 바로 적용해보겠습니다.